File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/documents/config/ServicesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.documents.config;

import io.zenwave360.example.clinicaltool.modules.documents.*;
import io.zenwave360.example.clinicaltool.modules.documents.domain.*;
import java.util.List;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

/**
 * Services InMemory Config. It can be used standalone or with @SpringBootTest.
 */
@Configuration
@Profile("in-memory")
public class ServicesInMemoryConfig extends RepositoriesInMemoryConfig {

    protected final DocumentServiceImpl documentService = new DocumentServiceImpl(documentInfoRepository());

    @Bean
    public <T extends DocumentService> T documentService() {
        return (T) documentService;
    }

    static List<DocumentInfo> _documentInfos;

    public void reloadTestData() {

        var testDataLoader = new TestDataLoader(List.of(DocumentInfo.class, DocumentData.class));
        var documentInfos = _documentInfos != null
                ? _documentInfos
                : testDataLoader.loadCollectionTestDataAsObjects(DocumentInfo.class);
        documentInfoRepository().deleteAll();
        documentInfoRepository().saveAll(documentInfos);
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/documents/config/ServicesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.documents.config


import io.zenwave360.example.clinicaltool.modules.documents.domain.*


import io.zenwave360.example.clinicaltool.modules.documents.*
import io.zenwave360.example.clinicaltool.modules.documents.*


    

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile

/**
 * Services InMemory Config. It can be used standalone or with @SpringBootTest.
 */
@Configuration
@Profile("in-memory")
open class ServicesInMemoryConfig : RepositoriesInMemoryConfig() {



    protected val documentService = DocumentServiceImpl(documentInfoRepository()
        )
    @Bean
    open fun <T : DocumentService> documentService(): T {
        return documentService as T
    }


	companion object {
		var _documentInfos: List<DocumentInfo>? = null
	}

	fun reloadTestData() {
        
		val testDataLoader = TestDataLoader(listOf(DocumentInfo::class.java, DocumentData::class.java))
		val documentInfos = _documentInfos ?: testDataLoader.loadCollectionTestDataAsObjects(DocumentInfo::class.java)
		documentInfoRepository().deleteAll()
		documentInfoRepository().saveAll(documentInfos)
	}
}
